planet/Objects.ocd/Environment.ocd/Cloud.ocd/Script.c

```diff
@@ -249,28 +280,136 @@ private func Precipitation()
 	return;
 }
 
-// Raindrop somewhere from the cloud.
-private func RainDrop()
+// Check if liquid is maybe in frozen form.
+private func RainMat()
 {
-	// Find Random Position.
-	var con = GetCon();
-	var wdt = GetDefWidth() * con / 500;
-	var hgt = GetDefHeight() * con / 700;
-	var x = RandomX(-wdt, wdt);
-	var y = RandomX(-hgt, hgt);
-	if (!GBackSky(x, y))
-		return false;
-	// Check if liquid is maybe in frozen form.
-	var mat;
 	if (rain_mat_freeze_temp != nil && GetTemperature() < rain_mat_freeze_temp)
-		mat = rain_mat_frozen;
+		return rain_mat_frozen;
 	else
-		mat = rain_mat;
-	// Create rain drop.	
-	CastPXS(mat, 1, 1, x, y);
+		return rain_mat;
+}
+
+local last_raindrop_color = -1;
```

Comment:
This should probably be initialized to `nil` instead of -1, because -1 (aka 0xFFFF'FFFF) is a valid color.